### PR TITLE
Introduce template feature (CSS, header/footer, license)

### PR DESCRIPTION
**Pull request by rockwillj**
I introduced the template feature in this plugin. It has the following mechanism.

1. layout.html consists of header.html, footer.html and layout.css
2. Use template files in project directory if exists. Then users can customize them if needed.
3. Use license template files with the same name as its license name
   - To support licenses that are not supported by default, such as mixed licenses, proprietary licenses, and minor licenses
   - (To use this mechanism, #56 needs to be merged)

### TODO

- [ ] Update `assets/licenses.html` and `assets/licenses.json`

### Structure Example

- plugin/src/main/resources/template/
  - licenses/
    - supported-license.html
  - layout.html
  - layout.css
  - header.html
  - footer.html
- app/template/ (created by a user)
  - licenses/
    - supported-license.html
    - mixed-license.html
    - proprietary-license.html
    - minor-license.html
  - layout.html
  - layout.css
  - header.html
  - footer.html

### CODE REVIEW

File: example/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -7,14 +7,6 @@
             android:orientation="vertical"
             >
 
-        <TextView
-                android:layout_width="wrap_content"
-                android:layout_height="wrap_content"
-                android:text="This library uses the following open source software:"
-                android:textAppearance="@style/TextAppearance.AppCompat.Medium"
-                android:padding="8dp"
-                />
```
Comment (rockwillj) [inline example/src/main/res/layout/activity_main.xml:16]: WebView should simply load only license.html, I think.

File: plugin/src/main/groovy/com/cookpad/android/licensetools/Templates.groovy [hunk 2]
```diff
@@ -47,12 +63,18 @@ public class Templates {
         return s.toString()
     }
 
-    static String readResourceContent(String filename) {
-        def templateFileUrl = Templates.class.getClassLoader().getResource(filename)
-        if (templateFileUrl == null) {
-            throw new FileNotFoundException("File not found: $filename")
+    static String readResourceContent(String filename, File projectDir) {
```
Comment (rockwillj) [inline plugin/src/main/groovy/com/cookpad/android/licensetools/Templates.groovy:66]: Passing `projectDir` is not good but I don't know the smart way...

File: plugin/src/main/resources/template/footer.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<div class="footer">
+  <p>This document is generated by <a href="https://github.com/cookpad/license-tools-plugin">Android License Tools Plugin</a>.</p>
```
Comment (rockwillj) [inline plugin/src/main/resources/template/footer.html:2]: Everyone must want to know what tool created this cool document 😇

File: plugin/src/main/resources/template/header.html [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<div class="header">
+  <p>This application uses the following open source software.</p>
+</div>
```
Comment (rockwillj) [inline plugin/src/main/resources/template/header.html:3]: You can write copyright as below if you want.

```html
<div class="header">
  <h1>Example App</h1>
  <p>Copyright &copy; 2017 <a href="http://example.com">Example.com</a>. All rights reserved.</p>
  <p>Example App is built using the following open source software.</p>
</div>
```

**Comment by gfx** [comment]
Great.

Can you add examples to use custom templates?

**Comment by rockwillj** [comment]
@gfx I added examples of custom template in 79a44e8.
I chose mixed license example. Because Glide license is `bsd_2_clauses` in default, but it must be `BSD, part MIT and Apache 2.0` actually.

**Comment by k4zy** [comment]
We are going to archive this repository.
Please migrate https://github.com/kazyApp/LicenseToolsPlugin
Wound you mind create this pull request to kazyApp/LicenseToolsPlugin.
Thank you for great pull request.

**Pull request closed without merging**